torch/optim/adamax.py

```diff
@@ -265,13 +267,22 @@ def _single_tensor_adamax(
     capturable: bool,
     has_complex: bool,
 ):
+    if capturable:
+        raise RuntimeError("capturable is not supported for single tensor Adamax (when foreach=False)")
 
     for i, param in enumerate(params):
         grad = grads[i]
         grad = grad if not maximize else -grad
         exp_avg = exp_avgs[i]
         exp_inf = exp_infs[i]
         step_t = state_steps[i]
+
+        # If compiling, the compiler will handle cudagraph checks, see note [torch.compile x capturable]
+        if not torch._utils.is_compiling() and capturable:
+            assert (
```

Comment:
nit: raise the proper error is nicer than plain assert

Reply:
I've removed this for now